projectmanagement/src/main/java/life/qbic/projectmanagement/application/sample/SampleRegistrationService.java

```diff
@@ -24,40 +25,33 @@ public class SampleRegistrationService {
 
     private final SampleCodeService sampleCodeService;
     private final SampleDomainService sampleDomainService;
+    private final ProjectInformationService projectInformationService;
 
     @Autowired
     public SampleRegistrationService(SampleCodeService sampleCodeService,
-                                     SampleDomainService sampleDomainService) {
+                                     SampleDomainService sampleDomainService,
+        ProjectInformationService projectInformationService) {
         this.sampleCodeService = Objects.requireNonNull(sampleCodeService);
         this.sampleDomainService = Objects.requireNonNull(sampleDomainService);
-    }
-
-    public Result<Sample, ResponseCode> registerSample(SampleRegistrationRequest sampleRegistrationRequest, ProjectId projectId) {
-        Objects.requireNonNull(sampleRegistrationRequest);
-        Objects.requireNonNull(projectId);
-        var sampleCode = sampleCodeService.generateFor(projectId);
-        if (sampleCode.isError()) {
-            return Result.fromError(ResponseCode.SAMPLE_REGISTRATION_FAILED);
-        }
-        var result = sampleDomainService.registerSample(sampleCode.getValue(), sampleRegistrationRequest);
-        if (result.isError()) {
-            return Result.fromError(ResponseCode.SAMPLE_REGISTRATION_FAILED);
-        }
-        return Result.fromValue(result.getValue());
+        this.projectInformationService = Objects.requireNonNull(projectInformationService);
     }
 
     public Result<Collection<Sample>, ResponseCode> registerSamples(
             Collection<SampleRegistrationRequest> sampleRegistrationRequests, ProjectId projectId) {
         Objects.requireNonNull(sampleRegistrationRequests);
         Objects.requireNonNull(projectId);
+        var project = projectInformationService.find(projectId);
+        if (project.isEmpty()) {
+            return Result.fromError(ResponseCode.SAMPLE_REGISTRATION_FAILED);
```

Comment:
since this will be ok for the user, it is hard to debug without logging it. I suggest to error log with project id, or you extend the response code